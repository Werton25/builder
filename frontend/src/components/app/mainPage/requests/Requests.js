import React from 'react';
import PropTypes from 'prop-types';
import Select from "./select/Select";
import {editRequest, saveRequest} from "../../../../actions/requests";
import {generateId} from "../../../../utils";
import config from "../../../../config";
import {connect} from 'react-redux';
import {isDefaultRequest} from "../../../../selectors/select";
import {getCurrentRequest} from "../../../../selectors/requests";
import {getCurrentUri} from "../../../../selectors/router";

export class BaseRequests extends React.Component {
  constructor(props) {
    super(props);
    this.state = {currentRequestName: this.props.currentRequest.name};
  }

  componentWillReceiveProps(nextProps) {
    this.setState({currentRequestName: nextProps.currentRequest.name});
  }

  render() {
    const {currentRequest, currentUri, isDefaultRequest, saveRequest, editRequest} = this.props;

    const {currentRequestName} = this.state;

    return (
      <div>
        <div className='col-md-4 col-sm-4'>
          <Select/>
        </div>
        <div className='col-md-4 col-sm-4'>
          <div className='input-group pull-right'>
            <input className='form-control' type='text' value={currentRequestName} placeholder='Введите имя набора'
                   onChange={(e) => this.setState({currentRequestName: e.target.value})}/>
            <div className='input-group-btn'>
              <button className='btn btn-default'
                      disabled={isDefaultRequest}
                      onClick={() => editRequest({
                        id: currentRequest.id,
                        uri: currentUri,
                        name: currentRequestName
                      })}>
                <i className='glyphicon glyphicon-pencil' title='Переименовать набор'/>
              </button>
              <button className='btn btn-default'
                      onClick={() => saveRequest({
                        id: generateId(),
                        uri: currentUri,
                        name: this.state.currentRequestName || config.defaultRequestName,
                        value: currentRequest.value
                      })}>
                <i className='glyphicon glyphicon-plus' title='Добавить набор'/>
              </button>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

BaseRequests.propTypes = {
  currentUri: PropTypes.string.isRequired,
  currentRequest: PropTypes.object.isRequired,
  isDefaultRequest: PropTypes.bool.isRequired,
  saveRequest: PropTypes.func.isRequired,
  editRequest: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({
  currentUri: getCurrentUri(state),
  currentRequest: getCurrentRequest(state),
  isDefaultRequest: isDefaultRequest(state)
});

const mapDispatchToProps = {
  saveRequest,
  editRequest
};

export default connect(mapStateToProps, mapDispatchToProps)(BaseRequests);
