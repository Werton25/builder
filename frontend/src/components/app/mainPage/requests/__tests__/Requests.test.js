import React from "react";
import renderer from 'react-test-renderer';
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import Requests from "../Requests";

const store = configureStore()({
  router: {location: {pathname: '/service1/method1'}},
  requests: {
    '/service1/method1': {
      'default': {
        id: 'default',
        name: 'default',
        value: 'test',
        schema: 'test'
      }
    }
  },
  select: {}
});

describe('Requests tests', () => {
  it('Requests renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Requests/>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
