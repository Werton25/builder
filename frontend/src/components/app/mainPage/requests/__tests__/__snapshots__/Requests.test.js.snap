// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Requests tests Requests renders correctly 1`] = `
<div>
  <div
    className="col-md-4 col-sm-4"
  >
    <div
      className="Select request-select has-value is-searchable Select--single"
      style={undefined}
    >
      <div
        className="Select-control"
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={undefined}
      >
        <span
          className="Select-multi-value-wrapper"
          id="react-select-2--value"
        >
          <div
            className="Select-value"
            style={undefined}
            title={undefined}
          >
            <span
              aria-selected="true"
              className="Select-value-label"
              id="react-select-2--value-item"
              role="option"
            >
              default
            </span>
          </div>
          <div
            className="Select-input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-activedescendant="react-select-2--value"
              aria-describedby={undefined}
              aria-expanded="false"
              aria-haspopup="false"
              aria-label={undefined}
              aria-labelledby={undefined}
              aria-owns=""
              className="Select-input"
              id={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              role="combobox"
              tabIndex={undefined}
              value=""
            />
          </div>
        </span>
        <span
          className="Select-arrow-zone"
          onMouseDown={[Function]}
        >
          <span
            className="Select-arrow"
            onMouseDown={[Function]}
          />
        </span>
      </div>
    </div>
  </div>
  <div
    className="col-md-4 col-sm-4"
  >
    <div
      className="input-group pull-right"
    >
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="Введите имя набора"
        type="text"
        value="default"
      />
      <div
        className="input-group-btn"
      >
        <button
          className="btn btn-default"
          disabled={true}
          onClick={[Function]}
        >
          <i
            className="glyphicon glyphicon-pencil"
            title="Переименовать набор"
          />
        </button>
        <button
          className="btn btn-default"
          onClick={[Function]}
        >
          <i
            className="glyphicon glyphicon-plus"
            title="Добавить набор"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
